tests/core/eth-module/test_accounts.py

```diff
@@ -0,0 +1,276 @@
+        web3.eth.account.privateKeyToAccount(b'\xff' * 33)
+
+
+def test_eth_account_privateKeyToAccount_properties(web3, PRIVATE_BYTES):
+    account = web3.eth.account.privateKeyToAccount(PRIVATE_BYTES)
+    assert callable(account.sign)
+    assert callable(account.signTransaction)
+    assert is_address(account.address)
+    assert account.address == '0xa79F6f349C853F9Ea0B29636779ae3Cb4E3BA729'
+    assert account.privateKey == PRIVATE_BYTES
+
+
+def test_eth_account_create_properties(web3):
+    account = web3.eth.account.create()
+    assert callable(account.sign)
+    assert callable(account.signTransaction)
+    assert is_address(account.address)
+    if sys.version_info.major < 3:
+        assert is_hex(account.privateKey) and len(account.privateKey) == 66
+    else:
+        assert isinstance(account.privateKey, bytes) and len(account.privateKey) == 32
+
+
+def test_eth_account_recover_transaction_example(web3):
+    raw_tx_hex = '0xf8640d843b9aca00830e57e0945b2063246f2191f18f2675cedb8b28102e957458018025a00c753084e5a8290219324c1a3a86d4064ded2d15979b1ea790734aaa2ceaafc1a0229ca4538106819fd3a5509dd383e8fe4b731c6870339556a5c06feb9cf330bb'  # noqa: E501
+    from_account = web3.eth.account.recoverTransaction(raw_tx_hex)
+    assert from_account == '0xFeC2079e80465cc8C687fFF9EE6386ca447aFec4'
+
+
+def test_eth_account_recover_transaction_with_literal(web3):
+    raw_tx = 0xf8640d843b9aca00830e57e0945b2063246f2191f18f2675cedb8b28102e957458018025a00c753084e5a8290219324c1a3a86d4064ded2d15979b1ea790734aaa2ceaafc1a0229ca4538106819fd3a5509dd383e8fe4b731c6870339556a5c06feb9cf330bb  # noqa: E501
+    from_account = web3.eth.account.recoverTransaction(raw_tx)
+    assert from_account == '0xFeC2079e80465cc8C687fFF9EE6386ca447aFec4'
+
+
+def test_eth_account_recover_message(web3):
+    v, r, s = (
+        28,
+        '0xe6ca9bba58c88611fad66a6ce8f996908195593807c4b38bd528d2cff09d4eb3',
+        '0x3e5bfbbf4d3e39b1a2fd816a7680c19ebebaf3a141b239934ad43cb33fcec8ce',
+    )
+    message = "I♥SF"
+    from_account = web3.eth.account.recoverMessage(text=message, vrs=(v, r, s))
+    assert from_account == '0x5ce9454909639D2D17A3F753ce7d93fa0b9aB12E'
+
+
+def test_eth_account_recover_signature_bytes(web3):
+    signature_bytes = b'\x0cu0\x84\xe5\xa8)\x02\x192L\x1a:\x86\xd4\x06M\xed-\x15\x97\x9b\x1e\xa7\x90sJ\xaa,\xea\xaf\xc1"\x9c\xa4S\x81\x06\x81\x9f\xd3\xa5P\x9d\xd3\x83\xe8\xfeKs\x1chp3\x95V\xa5\xc0o\xeb\x9c\xf30\xbb\x00'  # noqa: E501
+    msg_hash = b'\xbb\r\x8a\xba\x9f\xf7\xa1<N,s{i\x81\x86r\x83{\xba\x9f\xe2\x1d\xaa\xdd\xb3\xd6\x01\xda\x00\xb7)\xa1'  # noqa: E501
+    msg_hash = to_hex_if_py2(msg_hash)
+    signature = to_hex_if_py2(signature_bytes)
+    from_account = web3.eth.account.recover(msg_hash, signature=signature)
+    assert from_account == '0xFeC2079e80465cc8C687fFF9EE6386ca447aFec4'
+
+
+def test_eth_account_recover_vrs(web3):
+    v, r, s = (
+        27,
+        5634810156301565519126305729385531885322755941350706789683031279718535704513,
```

Comment:
Wondering if it would be good to source these test fixtures as well as maybe a small sample script for generating them from a known good implementation (*cough* `coincurve` *cough*).

This is more about demonstrating due-diligence than anything else.

Reply:
Fair enough. Off to learn me some `coincurve`...